src/gas/GasFeeController.ts

```diff
@@ -329,7 +329,9 @@ export class GasFeeController extends BaseController<
     this.EIP1559APIEndpoint = EIP1559APIEndpoint;
     this.legacyAPIEndpoint = legacyAPIEndpoint;
     this.clientId = clientId;
-    if (onNetworkStateChange && getChainId && getProvider) {
+    const initialProvider = getProvider();
+    this.ethQuery = new EthQuery(initialProvider);
```

Comment:
Shouldn't this be within the `if` block one line below? The `else` block creates its own instance

Reply:
yep

Reply:
moved [here](https://github.com/MetaMask/controllers/pull/936/commits/992d1f36270a2475945f5938f9e27799f24b0380)